feat: extend tabdata interface to include support for welcome screen

anasstahr opened the pull request:
## Problem
We don't have a way to integrate the current welcome screen in Q through flare

## Solution
Extending to protocol so we can integrate the welcome screen UI in flare

<!---
    REMINDER:
    - Read CONTRIBUTING.md first.
    - Add test coverage for your changes.
    - Link to related issues/commits.
    - Testing: how did you test your changes?
    - Screenshots if applicable
-->

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

volodkevych commented on types/chat.ts:
```diff
@@ -155,7 +158,10 @@ export interface QuickActions {
 }
 
 export interface TabData {
```
Should we add new properties to support setting tab header? (`tabHeaderDetails` property on vscode repo)

volodkevych replied:
I see you added `title`, but we still need `description` and header-specific `icon`. Shall we introduce this `tabHeader` as a new type in the protocol?

volodkevych commented on types/chat.ts:
```diff
@@ -155,7 +158,10 @@ export interface QuickActions {
+    title: string
     placeholderText?: string
+    placeHolderLabel?: string
+    centerContent?: boolean
```
Is it for `compactMode`?

anasstahr replied:
Yes, do we want to change the name to `compactMode` ?

volodkevych replied:
`compactMode` is sounds better for what that flag is doing

volodkevych commented on types/chat.ts:
```diff
@@ -155,8 +158,18 @@ export interface QuickActions {
 }
 
 export interface TabData {
+    title: string
```
Now, `title` is not needed here, right?

anasstahr replied:
you're right, missed that

volodkevych commented on types/chat.ts:
```diff
@@ -155,8 +158,18 @@ export interface QuickActions {
 export interface TabData {
+    title: string
     placeholderText?: string
+    placeHolderLabel?: string
```
nit: I did not know about `placeHolderLabel` when I added `placeholderText`. Maybe we could group those 2 props under `Placeholder` type. I know it's technically a breaking change, but nobody uses `TabData` yet, it was added recently, so I think we can still make this change without breaking any dependency.

anasstahr replied:
Sure that sounds good. Does the following sound ok?

```
placeHolder: PlaceHolder

export interface PlaceHolder{
   text?: string
   label?: string
}
```

volodkevych replied:
nit: `placeholder` should be 1 word, no camelCase?

anasstahr commented:
I changed the following after going through mynahUI again:
- Adding a signal for the welcome Counter that the vscode toolkit also uses to know when to update an internal state for the welcomeScreen
- Added the title back in `tabData`. This is different from the `tabHeader.title`. `tabData.title` is used for the actual title of the tab while `tabHeader.title` is used for text within the tab

justinmk3 commented on types/chat.ts:
```diff
@@ -155,10 +159,24 @@ export interface QuickActions {
+    description?: string
+}
+
+export interface PromptInput {
```
Why don't these have docstrings that give insight into how they should be used? Is that documented somewhere else? If not, can we make that a habit for all runtimes PRs?

viktorsaws commented on types/chat.ts:
```diff
@@ -100,6 +102,8 @@ export interface FileList {
 export interface ChatMessage {
     type?: 'answer' | 'prompt' | 'system-prompt' // will default to 'answer'
     body?: string
+    buttons: ChatItemAction[]
```
Why `buttons` are required field? It breaks compatibility in language-server repository, and code either should be adjusted there or this should be optional.

volodkevych replied:
FYI: this PR was reverted
